viziphant/unitary_event_analysis.py

```diff
@@ -0,0 +1,401 @@
+        identical to the one used to generate joint_surprise_dict.
+    window_size : quantities.Quantity
+        The size of the analysis-window. This value should be identical to the
+        one used to generate joint_surprise_dict.
+    window_step : quantities.Quantity
+        The size of the window step. This value should be identical to th one
+        used to generate joint_surprise_dict.
+    plot_params_user : dict
+        A dictionary of plotting parameters used to update the default plotting
+        parameter values.
+        Keys:
+        -----
+        events : dictionary (default: {})
+            Epochs to be marked on the time axis
+            key: epochs name as string
+            value: list of quantities.Quantity
+        figsize : tuple of int (default: (12, 10))
+            The dimensions for the figure size.
+        hspace : float (default: 1)
+            The amount of height reserved for white space between subplots.
+        wspace : float (default: 0.5)
+            The amount of width reserved for white space between subplots.
+        top : float (default: 0.95)
+        bottom : float (default: 0.1)
+        right : float (default: 0.87)
+        left : float (default: 0.08)
+            The sizes of the respective margin of the subplot in the figure.
+        fsize : integer (default: 12)
+            The size of the font
+        unit_real_ids : list of integers (default: [1, 2])
+            The unit ids form the experimental recording.
+        lw: float (default: 0.5)
+            The default line width.
+        S_ylim : tuple of ints or floats (default: (-3, 3))
+            The y-axis limits for the joint surprise plot.
+        marker_size : integers (default: 5)
+            The marker size for the coincidence and unitary events.
+        y_tick_interval : integers (default: 15)
+            The size of the interval between y-ticks (for rasterplots only)
+        time_unit : string (default: 'ms')
+            The time unit used to rescale the spiketrains.
+        frequency_unit : string (default: 'Hz')
+            The frequency unit used to rescale the spikerates.
+    Returns
+    -------
+    result : instance of namedtuple()
+        The container for Axis objects generated by this function. Individual
+        axes can be accessed using the respective identifiers:
+        result.identifier
+        Identifiers: spike_events_axis, spike_rates_axis,
+                     coincidence_events_axis, coincidence_rates_axis,
+                     statistical_significance_axis, unitary_events_axis
+    """
+    # update params_dict_default with user input
+    params_dict = params_dict_default.copy()
+    params_dict.update(plot_params_user)
+
+    # rescale all spiketrains to the uniform time unit from params_dict
+    for m in range(len(data)):
+        for n in range(len(data[0])):
+            data[m][n].rescale(params_dict['time_unit'])
```

Comment:
Rescale returns a copy of the data; it is not an in place operation. In other words, this line does nothing. You either need to save the new copy or immediately do an operation with the returned copy. 
See here: https://github.com/python-quantities/python-quantities/blob/master/quantities/quantity.py (lines 203-207)